adr/0026-schema.md

```diff
@@ -0,0 +1,102 @@
+
+  local:
+   path: chart
+```
+- `actions.wait` will be deprecated in favor of a wait list per component that is run on deploy. `action.wait` blocks that are `onDeploy.After` can be auto migrated and will work if deployed on v1. `wait` blocks that are not on `onDeploy.After` will not be auto migrated and the package will error out if deployed on v1. Zarf v1 will use kstatus for wait functionality instead of `kubectl wait`. `apiVersion` will be added as a required field on `wait`, and the `apiVersion` will be automatically set to the most recent `apiVersion` if the waited for resource is not a custom resource. Users can still wait on custom resources, but the resources must implement kStatus. `wait.condition` will be removed from the schema and Zarf will assume it is waiting for `ready`. A `wait` block will not be auto migrated if `condition` is not `set` to  `ready`, `available` or equivalent.
+In summary, a `wait` block will be auto migrated if, and only if:
+  - it is `onDeploy.after`
+  - the condition is `ready`, `available` or equivalent
+  - The kind is not a custom resource.
+- There may be further, more holistic changes to how actions works within Zarf that would have a significant affect on the schema. This will be covered in a separate ADR.
+
+### BDD scenarios
+The following are [behavior driven development](https://en.wikipedia.org/wiki/Behavior-driven_development) scenarios to provide context of what Zarf will do in specific situations given the above decisions.
+
+#### v1 create with deprecated keys
+- *Given* Zarf version is v1
+- *and* the `zarf.yaml` has no apiVersion or deprecated keys
+- *when* the user runs `zarf package create`
+- *then* they will receive an error and be told to run `zarf dev update-schema` or how to migrate off cosign key paths or how to use flavors over groups depending on the error
+
+#### v0 create -> v1 deploy
+- *Given*: A package is created with Zarf v0
+- *and* that package has deprecated keys that can be automatically migrated (required, scripts, & set variables)
+- *when* the package is deployed with Zarf v1
+- *then* the keys will be automatically migrated & the package will be deployed without error.
+
+#### v0 create with removed feature -> v1 deploy
+- *Given*: A package is created with Zarf v0
+- *and* that package has deprecated keys that cannot be automatically migrated (groups, cosignKeyPath)
+- *when* the package is deployed with Zarf v1
+- *then* then deploy of that package will fail and the user will be instructed to update their package
+
+#### v1 create -> v0 deploy
+- *Given*: A package is created with Zarf v1
+- *and* that package uses keys that don't exist in v0
+- *when* the package is deployed with Zarf v0
+- *then* Zarf v0 will deploy the package without issues. If there are fields unrecognized by the v0 schema, then the user will be warned they are deploying a package that has features that do not exist in the current version of Zarf.
```

Comment:
is this warning practically possible (it would need to be introduced - IIRC Zarf doesn't warn about extra keys today)

Reply:
Yeah it's planned to be introduced as a feature somewhat soon, but it will only work after it's introduced. I should note that in the ADR